File: 540. Single Element in a Sorted Array/Solution.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        n=len(nums)
        if nums[n-1]>nums[n-2]:
            return nums[n-1]
        start, end = 0, n-1
        while start<end:
            mid = (start+end)//2
            if mid%2==0 and nums[mid]==nums[mid+1]:
                start=mid+1
            elif mid%2==0 and nums[mid]!=nums[mid+1]:
                end=mid
            elif mid%2!=0 and nums[mid]!=nums[mid+1]:
                start=mid+1
            elif mid%2!=0 and nums[mid]==nums[mid+1]:
                end=mid-1
            
        return nums[start]

        

File: 540. Single Element in a Sorted Array/Solution.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int start=1,end=nums.size()-1;
        while(start<=end){
            int mid = start+(end-start)/2;
            if(mid%2==0){
                if(nums[mid]>nums[mid-1]){
                    start=mid+1;
                } else {
                     end=mid-1;
                }
            } else {
                 if(nums[mid]==nums[mid-1]){
                    start=mid+1;
                } else {
                    end=mid-1;
                }
            }
        }
        cout<<start<<"-"<<end<<endl;
        return nums[start-1];
    }
};